patternkit.module

```diff
@@ -69,26 +69,20 @@ function patternkit_flush_metadata_cache() {
  *   An array of Pattern Kit Library objects.
  */
 function patternkit_pattern_libraries() {
-  // @todo Add caching.
-  return module_invoke_all('patternkit_library');
-}
+  static $libraries;
 
-/**
- * Implements hook_patternkit_library().
- */
-function patternkit_patternkit_library() {
-  // @todo Replace with a Service that calls a Factory.
-  $rest_lib = new PatternkitRESTLib();
-  $libraries = array($rest_lib);
-  /** @var object $theme */
-  foreach (list_themes() as $theme_name => $theme) {
-    if ($theme->engine !== 'twig' || !isset($theme->info['namespaces'])) {
-      continue;
+  // If undefined, create the libraries metadata.
+  if (is_null($libraries)) {
```

Comment:
Let's use `if ($libraries === null) {` here.

Reply:
where are the coding standards you are following?

Reply:
https://mojo.redhat.com/docs/DOC-1133285 `is_null` is typically an unncessary function call, although I think PHP5.4+ inlines it before running